### ISSUE TITLE
🐛 React import checks are stricter than they should be

### ISSUE THREAD

**Issue by NotNite** [biomejs/biome#9020]
### Environment information

<details>

```bash
CLI:
  Version:                      2.3.14
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v25.2.1
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true
  HTML full support enabled:    unset

Workspace:
  Open Documents:               0
```

</details>

### What happened?

[is_global_react_import](https://github.com/biomejs/biome/blob/9d4cfa304268579361639c3d2301f1e38526dd00/crates/biome_js_analyze/src/react.rs#L305) is used by `lint/correctness/noUnusedImports` and `lint/style/useImportType` to not accidentally mark the React global as unused, as the classic React JSX transform requires it in scope. However, the function unnecessarily checks the import name against [`ReactLibrary::import_names`](https://github.com/biomejs/biome/blob/9d4cfa304268579361639c3d2301f1e38526dd00/crates/biome_js_analyze/src/react.rs#L122) to verify that it's actually React, and this behavior doesn't align with the rest of the ecosystem ([TypeScript](https://github.com/microsoft/TypeScript/blob/cdb583a81357553fd72faedc6602c3f2516490fc/src/services/organizeImports.ts#L365-L369
) simply accepts any import with the identifier of "React").

A project of mine [re-exports React](https://github.com/moonlight-mod/moonlight/blob/becf5cb3d182bf2f1d68700bbe29e5edb0a04035/packages/mappings/src/mappings/react.ts#L3-L6), so `is_global_react_import` returns false, and these lints don't work properly as a result:

```tsx
import React from "@moonlight-mod/wp/react";

export function Test() {
  return <span>Hello, world!</span>;
}
```

```
packages/core-extensions/src/test.tsx:1:8 lint/correctness/noUnusedImports  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ⚠ This import is unused.

  > 1 │ import React from "@moonlight-mod/wp/react";
      │        ^^^^^
    2 │
    3 │ export function Test() {

  ℹ Unused imports might be the result of an incomplete refactoring.

  ℹ Unsafe fix: Remove the unused imports.

    1 │ import·React·from·"@moonlight-mod/wp/react";
      │ --------------------------------------------

Checked 1 file in 7ms. No fixes applied.
Found 1 warning.
```

If I use React as a type (e.g. set the return type to `React.ReactNode`), `useImportType` kicks in. This is arguably worse, as it contains a "safe fix" that prevents the code from compiling:

```
$ biome check
packages/core-extensions/src/test.tsx:1:8 lint/style/useImportType  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ⚠ All these imports are only used as types.

  > 1 │ import React from "@moonlight-mod/wp/react";
      │        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    2 │
    3 │ export function Test(): React.ReactNode {

  ℹ Importing the types with import type ensures that they are removed by the compilers and avoids loading unnecessary modules.

  ℹ Safe fix: Use import type.

    1 │ import·type·React·from·"@moonlight-mod/wp/react";
      │        +++++

Checked 1 file in 10ms. No fixes applied.
Found 1 warning.
```

```
$ tsc --noEmit
src/test.tsx:4:11 - error TS1361: 'React' cannot be used as a value because it was imported using 'import type'.

4   return <span>Hello, world!</span>;
            ~~~~

  src/test.tsx:1:8
    1 import type React from "@moonlight-mod/wp/react";
             ~~~~~~~~~~
    'React' was imported here.


Found 1 error in src/test.tsx:4
```

### Expected result

`is_global_react_import` shouldn't enforce the import name, to match behavior with the rest of the ecosystem. This is also how esbuild and [@typescript-eslint/no-unused-vars](https://typescript-eslint.io/rules/no-unused-vars/) functions, but I'm not entirely sure what the responsible code is in those projects.

Alternatively, some way to configure or disable this behavior would be nice. Judging by https://github.com/biomejs/biome/pull/7488, this also seems to have caused problems for [@rbxts/react](https://github.com/littensy/rbxts-react) in the past, however that PR simply added the package name to the hardcoded list. "I do not foresee a significant amount of users otherwise ever having to modify this behavior to point to a separate package"... oops.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by siketyan** [comment]
See https://biomejs.dev/reference/configuration/#javascriptjsxruntime

**Comment by NotNite** [comment]
> See [biomejs.dev/reference/configuration#javascriptjsxruntime](https://biomejs.dev/reference/configuration/#javascriptjsxruntime)

This happens with `jsxRuntime` set to `reactClassic`.

**Comment by siketyan** [comment]
My bad, using re-exported React or React-like libraries is a use-case we don't support yet. It's similar to https://github.com/biomejs/biome/issues/3682 but this is for custom runtime modules.

**Comment by ematipico** [comment]
I believe we landed this feature in the next branch @siketyan

**Comment by NotNite** [comment]
I've created a repro project: <https://github.com/NotNite/biome-9020-repro>. Biome will lint the React import as unused, but tsc will transform it fine anyways, which confirms that the library name check is excessive:

```js
// dist/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Test = Test;
const not_actually_react_1 = __importDefault(require("./not-actually-react"));
function Test() {
    return not_actually_react_1.default.createElement("span", null, "Hello, world!");
}
```

> I believe we landed this feature in the next branch [@siketyan](https://github.com/siketyan)

If you mean https://github.com/biomejs/biome/commit/e90b14f0e6789b4f402bf9621dfa322ebed565ea, that's similar but unrelated. The only JSX related setting in `tsconfig.json` is `"jsx": "react"`. This should work out of the box with zero extra configuration required (beyond `"jsxRuntime": "reactClassic"`), as the semantics are exactly the same as the classic React JSX factory.

Also, this isn't a feature, it's a bug? I'm not sure why this was marked as enhancement.

**Comment by siketyan** [comment]
Yeah I could still reproduce it in `next`: https://next.biomejs.dev/playground/?lintRules=noUnusedImports&files.main.tsx=aQBtAHAAbwByAHQAIABSAGUAYQBjAHQAIABmAHIAbwBtACAAIgByAGUAYQBjAHQAIgA7AAoAaQBtAHAAbwByAHQAIABSAGUAYQBjAHQAIABmAHIAbwBtACAAIgAuAC8AbgBvAHQALQBhAC0AcgBlAGEAYwB0ACIAOwAKAGkAbQBwAG8AcgB0ACAARgBvAG8AIABmAHIAbwBtACAAIgAuAC8AcwBvAG0AZQB0AGgAaQBuAGcAIgA7AAoA&files.biome.json=ewAKACAAIAAiAGYAaQBsAGUAcwAiADoAIAB7AAoAIAAgACAAIAAiAGkAbgBjAGwAdQBkAGUAcwAiADoAIABbACIAKgAqACIAXQAKACAAIAB9ACwACgAgACAAIgBsAGkAbgB0AGUAcgAiADoAIAB7AAoAIAAgACAAIAAiAGUAbgBhAGIAbABlAGQAIgA6ACAAdAByAHUAZQAKACAAIAB9ACwACgAgACAAIgBqAGEAdgBhAHMAYwByAGkAcAB0ACIAOgAgAHsACgAgACAAIAAgACIAagBzAHgAUgB1AG4AdABpAG0AZQAiADoAIAAiAHIAZQBhAGMAdABDAGwAYQBzAHMAaQBjACIACgAgACAAfQAKAH0ACgA%3D

**Comment by siketyan** [comment]
> Also, this isn't a feature, it's a bug? I'm not sure why this was marked as enhancement.

At a grance I thought this is a use-case that Biome doesn't support yet, but turns out it was a mismatch of the behaviour between our logic and tsc. I'll revert to mark it as a bug.
